src/transformers/trainer.py

```diff
@@ -894,3 +951,49 @@ def distributed_concat(self, tensor: torch.Tensor, num_total_examples: int) -> t
         # truncate the dummy elements added by SequentialDistributedSampler
         output = concat[:num_total_examples]
         return output
+
+    def eval_step(
```

Comment:
I got a bit lost with the naming: both `evaluate` and `predict` call `prediction_loop` which call `eval_step`.
Maybe rename it to `prediction_step`?
I was a bit confused between what is for evaluation and what is for prediction.
Feel free to ignore if you think it's far-fetched.

Reply:
Yes, `prediction_step` makes more sense since it's called by `prediction_loop`. Will adjust.